mkosi/__init__.py

```diff
@@ -384,8 +380,8 @@ def format_bytes(num_bytes: int) -> str:
     return f"{num_bytes}B"
 
 
-def roundup512(x: int) -> int:
-    return (x + 511) & ~511
+def roundup(x: int, step: int) -> int:
+    return (x + step-1) & ~(step-1)
```

Comment:
hmm, `& ~(step-1)` only works if `step` is an exponent of 2, but the very generic name doesn't make this clear. maybe write as:

```py
return ((x + step - 1) // step) * step 
```

which has the big benefit that it works for any values of `step`